examples/custom-server/src/server.ts

```diff
@@ -0,0 +1,59 @@
+import dotenv from 'dotenv'
+import next from 'next'
+import nextBuild from 'next/dist/build'
+import path from 'path'
+
+dotenv.config({
+  path: path.resolve(__dirname, '../.env'),
+})
+
+import express from 'express'
+import payload from 'payload'
+
+import { seed } from './seed'
+
+const app = express()
+const PORT = process.env.PORT || 3000
+
+const start = async (): Promise<void> => {
+  await payload.init({
+    secret: process.env.PAYLOAD_SECRET || '',
+    mongoURL: process.env.MONGODB_URI || '',
+    express: app,
+    onInit: () => {
+      payload.logger.info(`Payload Admin URL: ${payload.getAdminURL()}`)
+    },
+  })
+
+  if (process.env.PAYLOAD_SEED === 'true') {
+    payload.logger.info('---- SEEDING DATABASE ----')
+    await seed(payload)
+  }
+
+  if (process.env.NEXT_BUILD) {
+    app.listen(PORT, async () => {
+      payload.logger.info(`Next.js is now building...`)
+      // @ts-expect-error
+      await nextBuild(path.join(__dirname, '../'))
+      process.exit()
+    })
+  }
+
+  const nextApp = next({
+    dev: process.env.NODE_ENV !== 'production',
+  })
+
+  const nextHandler = nextApp.getRequestHandler()
+
+  app.get('*', (req, res) => nextHandler(req, res))
+
+  nextApp.prepare().then(() => {
+    payload.logger.info('Next.js started')
+
+    app.listen(PORT, async () => {
+      payload.logger.info(`Server listening on ${PORT}...`)
+    })
+  })
+}
+
+start()
```

Comment:
Here is a way to avoid the "Connection refused" error everytime the server restarts
```suggestion
  nextApp.prepare().then(() => {
    payload.logger.info('Next.js started')
  })
}

const started = start()

// Hold off on serving requests until Payload is ready and Next.js is mounted
app.get('*', async (req, res, next) => {
  await started
  next()
})

app.listen(PORT, async () => {
  payload.logger.info(`Server listening on http://localhost:${PORT}...`)
})
```

Reply:
Ok is this in relation to https://github.com/payloadcms/nextjs-custom-server/issues/38? How can I reproduce?

Reply:
Yes idd, related to #38

Repro:
1. Start dev server under nodemon
2. Open a nextjs page
3. Make a file change
4. Focus the nextjs page before the server is restarted -> this gives the error sometimes

Reply:
Hmm, I was not able to get your change to work because `next()` is undefined where you have it. If you're able to clone this down and get it in a working state, that would be amazing!